plugin-server/src/utils/db/db.ts

```diff
@@ -965,36 +965,43 @@ export class DB {
         return insertResult.rows[0]
     }
 
-    // Feature Flag Hash Key overrides
-    public async addFeatureFlagHashKeysForMergedPerson(
+    public async updateCohortsAndFeatureFlagsForMerge(
         teamID: Team['id'],
         sourcePersonID: Person['id'],
         targetPersonID: Person['id'],
         tx?: TransactionClient
     ): Promise<void> {
-        // Delete and insert in a single query to ensure
-        // this function is safe wherever it is run.
-        // The CTE helps make this happen.
-        //
-        // Every override is unique for a team-personID-featureFlag combo.
-        // In case we run into a conflict we would ideally use the override from most recent
-        // personId used, so the user experience is consistent, however that's tricky to figure out
-        // this also happens rarely, so we're just going to do the performance optimal thing
-        // i.e. do nothing on conflicts, so we keep using the value that the person merged into had
+        // When personIDs change, update places depending on a person_id foreign key
+
         await this.postgres.query(
             tx ?? PostgresUse.COMMON_WRITE,
-            `
-            WITH deletions AS (
-                DELETE FROM posthog_featureflaghashkeyoverride WHERE team_id = $1 AND person_id = $2
+            // Do two high level things in a single round-trip to the DB.
+            //
+            // 1. Update cohorts.
+            // 2. Update (delete+insert) feature flags.
+            //
+            // NOTE: Every override is unique for a team-personID-featureFlag combo. In case we run
+            // into a conflict we would ideally use the override from most recent personId used, so
+            // the user experience is consistent, however that's tricky to figure out this also
+            // happens rarely, so we're just going to do the performance optimal thing i.e. do
+            // nothing on conflicts, so we keep using the value that the person merged into had
+            `WITH cohort_update AS (
+                UPDATE posthog_cohortpeople
+                SET person_id = $1
+                WHERE person_id = $2
```

Comment:
Would it help to have a team filter here also?

Reply:
Looks like no `Team` here (it's just implied by the `Person` FK): https://github.com/PostHog/posthog/blob/d77858d275e0cc03ac5ea429eb81334cab7af22e/posthog/models/cohort/cohort.py#L336-L343